src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,256 @@
+        /// </summary>
+        /// <param name="uartDevice">Path to the UART device / serial port, e.g. /dev/serial0</param>
+        /// <exception cref="System.ArgumentException">uartDevice is null or empty</exception>
+        public Mhz19b(string uartDevice)
+        {
+            if (string.IsNullOrEmpty(uartDevice))
+            {
+                throw new ArgumentException(nameof(uartDevice));
+            }
+
+            // create serial port using the setting acc. to datasheet, pg. 7, sec. general settings
+            _serialPort = new SerialPort(uartDevice, 9600, Parity.None, 8, StopBits.One)
+            {
+                Encoding = Encoding.ASCII,
+                ReadTimeout = 1000,
+                WriteTimeout = 1000
+            };
+            _serialPort.Open();
+            _serialPortStream = _serialPort.BaseStream;
+            _shouldDispose = true;
+        }
+
+        /// <summary>
+        /// Gets the current CO2 concentration from the sensor.
+        /// The validity is true if the current concentration was successfully read.
+        /// If the serial communication timed out or the checksum was invalid the validity is false.
+        /// If the validity is false the ratio is set to 0.
+        /// </summary>
+        /// <returns>CO2 concentration in ppm and validity</returns>
+        /// <exception cref="System.IO.IOException">Communication with sensor failed</exception>
+        public VolumeConcentration GetCo2Reading()
+        {
+            try
+            {
+                // send read command request
+                var request = CreateRequest(Command.ReadCo2Concentration);
+                request[(int)MessageFormat.Checksum] = Checksum(request);
+                _serialPortStream.Write(request, 0, request.Length);
+
+                // read complete response (9 bytes expected)
+                byte[] response = new byte[MessageSize];
+                int timeout = 100;
+                int bytesRead = 0;
+                while (timeout > 0 && bytesRead < MessageSize)
+                {
+                    bytesRead += _serialPortStream.Read(response, bytesRead, response.Length - bytesRead);
+                    Thread.Sleep(1);
+                    timeout--;
+                }
+
+                if (timeout == 0)
+                {
+                    throw new IOException("Timeout");
+                }
+
+                // check response and return calculated concentration if valid
+                if (response[(int)MessageFormat.Checksum] == Checksum(response))
+                {
+                    return VolumeConcentration.FromPartsPerMillion((int)response[(int)MessageFormat.DataHighResponse] * 256 + (int)response[(int)MessageFormat.DataLowResponse]);
+                }
+                else
+                {
+                    throw new IOException("Invalid response message received from sensor");
+                }
+            }
+            catch (Exception e)
```

Comment:
you should catch only what you expect, otherwise you should let anything throw as it was. Anything specific you're guarding against?

Reply:
Well, no. I just wanted to hide all the details from user. In any case the communication with the sensor caused the exception therefore I mapped them all to IOException to keep the exceptions lean.
Alternatively I could remove the try...catch completely. How do you handle the documentation in this case? Add a description for each possible exception?

Reply:
I removed the try...catch so that any exception will passed trough as is.
Done.